### PR TITLE
[Notifications] feat: toggle notifications muted for a colony

### LINKED ISSUE #3005: Notifications: Allow users to toggle notifications for a colony
### Story

As a *user who is in multiple colonies*, I want to *be able to toggle notifications on a colony level*, so that *I can control which colonies I receive notifications from*

### Description

We want to give users the option to turn off notifications for a specific colony, since they might want to only receive notifications from one, but not another. This is done via the kebab menu inside a colony context. We need to store these colony specific preferences inside the notifications data object on the user in our db. This ticket should cover all screen sizes.

### Design

[Figma link](https://www.figma.com/design/0Fi3AaDlr3LAXBfsOXBGyk/User-Account?node-id=6401-23041&t=RpPiEaurGEYf4K5z-4)

Image

### This issue is complete when...

- [ ] The user can see and click an option to mute / unmute notifications for a specific colony.
- [ ] These preferences are stored in our db.

### Suggestions

* My idea was to create an array inside the notifications data called something like 'mutedColonyIDs' which would contain a list of the IDs of all of the muted colonies. When you mute a colony, it would add it's ID to this array, and when you unmute, it would remove it from the array. Later when we start sending notifications we'll need to check this array and ensure that the colony the notification is coming from is not in this list before sending.

### PR DESCRIPTION
**Pull request by bassgeta**
## Description

Added the toggle button in the dropdown menu on the colony homepage.
`DropdownMenu` needs to get refactored since for displaying that menu item, I had to create a **hook** :tada: which returns config data depending on the muted status and loading status.  Kinda overhead.

[Auth proxy PR](https://github.com/JoinColony/colonyCDappAuthProxy/pull/28)

## Testing
Before testing this, ping me or @davecreaser to give you the env secrets in order to test this.
Create a new .env file in the `createUserNotificationsData` with 
```
MAGICBELL_API_KEY=
MAGICBELL_API_SECRET=
```

1. Open up the app as `leela` (or any user), run the following query to assure you don't have any muted colonies yet:
```
query MyQuery {
  getUser(id: "0xb77D57F4959eAfA0339424b83FcFaf9c15407461") {
    id
    profile {
      displayName
    }
    notificationsData {
      magicbellUserId
      mutedColonyIds
      userAddress
    }
  }
}
```
image
2. Throttle your network to 3G, open the dropdown menu to the right and notice the `Mute notifications` item. Click it, it should change into `Muting notifications` and later into `Unmute notifications`. THe button should stay disabled throughout the entire ordeal.
image
image
image
3. Run the query to be sure it's added to the backend
image
4. Do it the other way around now!
image
image
image
5. Run the query and be sure the array is now empty
image

## Diffs

**New stuff** ✨

* `useMuteItem` hook for displaying the muted item in a smaller hook
* `mutedColonyIds` to the user notification object

**Changes** 🏗

* `mutedColonyIds` to `NotificationContext`

Resolves  #3005

### CODE REVIEW

File: src/components/v5/frame/ColonyHome/partials/DashboardHeader/partials/ColonyLinks/useMuteColonyItem.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        icon: BellSimpleSlash,
+        disabled: true,
+      };
+    }
+
+    return {
+      key: ITEM_KEY,
+      label: formatText(MSG.unmute),
+      icon: BellRinging,
+      onClick: () => {
```
Comment (davecreaser) [inline src/components/v5/frame/ColonyHome/partials/DashboardHeader/partials/ColonyLinks/useMuteColonyItem.ts:62]: Might be nice to make this and the other onClick into named functions wrapped in useCallback.

**Review by davecreaser — APPROVED** [review]
Great stuff @bassgeta! Good solution with the hook 😆 

Just left one suggestion in a comment, but feel free to not do it if you don't want.

Tested and it's buttery smooth:

Screenshot 2024-09-03 at 16 19 16
Screenshot 2024-09-03 at 16 22 35
Screenshot 2024-09-03 at 16 22 50
Screenshot 2024-09-03 at 16 22 56
Screenshot 2024-09-03 at 16 23 02

**Review by iamsamgibbs — APPROVED** [review]
Great work on this! Tested without issue.

Screenshot 2024-09-03 at 16 49 42

Screenshot 2024-09-03 at 16 49 55

Screenshot 2024-09-03 at 16 50 03

Screenshot 2024-09-03 at 16 50 12

Screenshot 2024-09-03 at 16 53 27

Also tested with multiple colonies:

Screenshot 2024-09-03 at 16 53 48

Will be great to try this out with some actual notifications!

**Review by rdig — APPROVED** [review]
Yeah, all good here. Tested, muted, unmuted. Everything works.

The only thing that I couldn't get to work is the `.env` file inside the lambda. I had to put the api keys/secrets in the lambda itself. But that might be just something on my end.

Screenshot from 2024-09-11 12-53-21
Screenshot from 2024-09-11 13-23-40
Screenshot from 2024-09-11 13-23-48
Screenshot from 2024-09-11 13-24-28

**Pull request merged**